### ISSUE TITLE
issue of the workflow for similarity graph build and featrue extraction

### ISSUE THREAD

**Issue by json87** [AIBluefisher/GraphSfM#55]
Hi, thanks for providing the toolkit. I find that the workflow for similarity graph build and feature extraction is not correct as features are required for vocabulary tree-based image retrieval. When the database is empty, the step 1 does nothing.

![image](https://user-images.githubusercontent.com/17582213/148745759-824229ee-f300-4796-bcae-b4a9c3ec0b43.png)

**Comment by AIBluefisher** [comment]
Yes, you're right. If we use the internal similarity computation method, we must perform feature extraction first.
